torchqdynamics/solvers/solver.py

```diff
@@ -24,9 +24,12 @@ def H_squared(self, t):
     @functools.wraps(func)
     def wrapper(instance, t, *args, **kwargs):
         if func.__name__ not in instance._cache or (
-            t != instance._cache[func.__name__][0] and instance._H_changed(t)
+            (t, args, kwargs) != instance._cache[func.__name__][0]
```

Comment:
There is a topic here, because args and kwargs can be different in value but have `old_args == args` if args contain pointer to objects that have been modified. IMO this is very misleading can entail non trivial bugs.
To solve it, two solutions : 
- perform a deep comparison of the arguments (cf all the machinery in https://github.com/pierreguilmin/torchqdynamics/blob/abocquet/time-dependent-hamiltonian-2/torchqdynamics/decorators.py#L50)
- put the dust under the rug and only support 't'. This sounds reasonable as I don't see a usecase for calling twice a H dependant function with the same time but different arguments, but maybe I'm missing something

Reply:
I agree that this modification may lead to very subtle bugs. I would favour the second cheap proposed solution (only care about `t`) as discussed at the time.

Reply:
Ok that's fine with me. I will add a warning in the docstring then.